src/main/java/Duke.java

```diff
@@ -1,10 +1,272 @@
+        }
+        else if (userInput.startsWith("todo ")) {
+            storedTasks[additions] = new Todo(userInput);
+        }
+
+        printAddedResponse(storedTasks[additions]);
+        additions++;
+    }
+
+    /**
+     * Executes Task mode, where user's commands are added into to-do list.
+     * When "list" entered, prints out current to-do list.
+     * When "completed <task id>" or "done <task id>" entered, updates that task status.
+     * When "remove <task id>" or "clear <task id>" entered, removes that task from list.
+     * When "change" is typed in by user, switches program to Echo mode.
+     */
+    public static void startTask() {
+        Scanner in = new Scanner(System.in);
+        String toAdd = in.nextLine();
+        String textLowerC = toAdd.toLowerCase();
+
+        while (!textLowerC.equals("bye") & !textLowerC.equals("list") & !textLowerC.equals("change")) {
+            defaultTaskModeBehavior(toAdd, textLowerC);
+            toAdd = in.nextLine();
+            textLowerC = toAdd.toLowerCase();
+        }
+
+        if (textLowerC.equals("list")) {
+            printList();
+            startTask();
+        }
+
+        // Switch to Echo mode if user types in "change"
+        if (textLowerC.equals("change")) {
+            switchMode(ECHO_MODE);
+        }
+    }
+
+    /**
+     * Checks different scenarios for input commands of user in Task mode and
+     * does the necessary action.
+     *
+     * @param toAdd String input by user.
+     * @param textLowerC String input by user converted to lower cases for checking.
+     */
+    private static void defaultTaskModeBehavior(String toAdd, String textLowerC) {
+        // Mark task as complete with an X.
+        if (textLowerC.startsWith("completed ") | textLowerC.startsWith("done ")) {
+            int taskNo = Integer.parseInt(toAdd.replaceAll("[^0-9]", "")) - 1;
+            storedTasks[taskNo].setDone();
+            // Print response when task marked done.
+            System.out.println("\t" + HOR_LINE);
+            System.out.printf("\tThat's great! %s has been checked as completed!\n", storedTasks[taskNo].description);
+            System.out.println("\t" + HOR_LINE + System.lineSeparator());
+        }
+        // Remove task from list.
+        else if (textLowerC.startsWith("clear ") | textLowerC.startsWith("remove ")) {
+            int taskNo = Integer.parseInt(toAdd.replaceAll("[^0-9]", "")) - 1;
+            System.out.println("\t" + HOR_LINE);
+            System.out.printf("\t%s removed from list!\n", storedTasks[taskNo].description);
+            System.out.println("\t" + HOR_LINE + System.lineSeparator());
+            System.arraycopy(storedTasks,taskNo + 1, storedTasks, taskNo, additions - taskNo);
+            additions--;
+            printList();
+        }
+        // Create a new task if it does not exist in list
+        else {
+            createTask(toAdd);
+        }
```

Comment:
As per the coding standard, perhaps you can remove the newline before else? (https://se-education.org/guides/conventions/java/basic.html#layout) Noticed this in a couple other places too.

Reply:
Nice catch. Will rectify in the next commit. :)